### PR TITLE
Upgrade to .net 9.0 preview

### PR DESCRIPTION
**Pull request by LightYagami28**
Upgrade all code to .net 9.0 preview for modern syntax

### CODE REVIEW

**Review by alphaO4 — COMMENTED** [review]
No malicious code added.
If the update actually works or it's "updated" by chatGPT remains to be seen.

**Comment by LightYagami28** [comment]
It's an update made at place with visual studio 2022 and Net 9.0 preview i also made future upgrade to other file

**Comment by LightYagami28** [comment]
Remove Deprecated ILMerge Package

Motivation: The ILMerge package (version 3.0.41) has been deprecated and is no longer supported by Microsoft. This package is no longer compatible with newer versions of the .NET Framework, which can lead to potential issues and compatibility problems.

Reason for Removal: To ensure the solution remains compatible with newer versions of the .NET Framework and to avoid any potential issues that may arise from using a deprecated package, we have removed the ILMerge package from all projects in the solution.

Affected Projects: The following projects have been updated to remove the ILMerge package:

    Extra
    FileManager
    LimeLogger
    Miscellaneous
    Options
    ProcessManager
    RemoteCamera
    RemoteDesktop
    SendFile
    SendMemory

Changes: The ILMerge package has been removed from the packages.config file and the package folder.

Benefits: This change ensures the solution remains compatible with newer versions of the .NET Framework, reducing the risk of potential issues and compatibility problems.

Testing: The solution has been tested to ensure that the removal of the ILMerge package does not affect the functionality of the projects.

**Comment by LightYagami28** [comment]
Migration to net sdk 9.0 to net framework 4.8

File: AsyncRAT-C#/Plugin/Chat/Chat/Connection.cs
Comment (alphaO4) [inline AsyncRAT-C#/Plugin/Chat/Chat/Connection.cs:1]: Is there a particular reason why you completly removed Connection.cs?

Comment (LightYagami28) [reply AsyncRAT-C#/Plugin/Chat/Chat/Connection.cs:1]: I also made a rewrite all entire code for net sdk

**Comment by LightYagami28** [comment]
# Changelog

- **Updated to .NET SDK 9.0**: Refactored the project to fully utilize the features and conventions of the latest .NET SDK 9.0.

- **Replaced `SslProtocols.Tls` with `SslProtocols.Tls12`**: Ensured the use of a more secure and up-to-date TLS version (1.2) for enhanced security.

- **Concurrency management improvements**: 
  - Replaced `lock (_sendSync)` with `SemaphoreSlim` for more efficient handling of concurrency.
  - Added structured error handling with `try-catch` blocks to enhance code stability and robustness.

- **Refactored method signatures**: 
  - Added `Task` return types to `InitializeClientAsync`, `SendAsync`, `CheckServer`, and `ReadServerDataAsync` to support asynchronous programming practices.
  - Added `bool` return type to `ValidateServerCertificate` for clearer method contracts.
  - Added `Task` return type to `ReadServerDataAsync` and `CheckServer` methods.

- **Optimized `SslStream.WriteAsync` calls**:
  - Replaced `await _sslClient.WriteAsync(buffersize)` with `await _sslClient.WriteAsync(buffersize.AsMemory())` to leverage the correct overload and improve performance.
  - Replaced `await _sslClient.WriteAsync(chunk)` with `await _sslClient.WriteAsync(chunk.AsMemory(0, read))` for accurate handling of chunked data transfers.

- **Simplified variable declarations**: Replaced explicit `new byte[4]` with `var buffer = new byte[4]`, enhancing code readability and reducing redundancy.

- **Enhanced networking logic**: 
  - Refined the usage of `TcpClient` and `SslStream` classes for establishing and managing secure connections with the server, following best practices for socket communication.
  - Replaced `X509Certificate` with `X509Certificate2` to utilize the more recent class, ensuring better compatibility and functionality.

- **Added null checks**: Implemented nullability checks for `_serverCertificate` and `_tcpClient.Client.RemoteEndPoint.ToString()` to prevent null reference exceptions.

- **General improvements**: 
  - Replaced the public property `IsConnected` with a private field `_isConnected` to avoid potential race conditions and ensure thread safety.
  - Substituted the public `Hwid` property with a private field `_hwid`, improving encapsulation and mitigating concurrency issues.

**Comment by LightYagami28** [comment]
I announce the closure of the pull request due to the extreme difficulty in completely rewriting the code in net 9

**Pull request closed without merging**